### ISSUE TITLE
TransitionTo deletes the character "?"

### ISSUE THREAD

**Issue by MrEfrem** [rackt/react-router#1070]
I transition from 
```
http://my/test/
```
to
```
http://my/test/?a=b
```
but i receive
```
http://my/test/a=b
```

I proxied
```
http://my/test/
```
to 
```
http://localhost:3000/test/
```
and any parameters shall be transferred after a slash.

**Comment by srph** [comment]
A little clarification -- do you mean `http://localhost:3000/test/?a=b` works, but not `http://my/test/a=b`?Mind including your code for the routes?

**Comment by MrEfrem** [comment]
Now works so
```
TransitionTo(http://my/test/?a=b) = http://my/test/a=b.
```

And it is necessary to me that worked so
```
TransitionTo(http://my/test/?a=b) = http://my/test/?a=b.
```

**Comment by MrEfrem** [comment]
Works URL
```
http://my/test/
```

Don't works URL
```
http://my/test
```

Therefore it is necessary to me
```
http://my/test/?a=b
```
but not
```
http://my/test?a=b
```

**Comment by MrEfrem** [comment]
Works if to specify "?" two times one after another. But it is more haque, than the correct decision.
```
http://my/test/??a=b
```

**Comment by idolizesc** [comment]
Yes, I noticed this as well.

`router.replaceWith('something/?a=b')` will result in `something/a=b` **(INCORRECT)**, while `router.replaceWith('something?a=b')` will result in `something?a=b` **(CORRECT)**.

Both should technically be the same, but the trailing slash seems to mess it up currently.

A temporary hack is to remove the trailing slash in your code.

**Comment by MrEfrem** [comment]
You didn't understand. I intentionally add trailing slash. Because I need to proxy the requests beginning on http://my/test/ and with trailing slash.

**Comment by ryanflorence** [comment]
can anybody confirm this for the code on master?

**Comment by mjackson** [comment]
I think this is the same thing as #1148

**Comment by ryanflorence** [comment]
the new history lib that router is built on fixes this, as seen in rackt/history@d6380e5

**Comment by idolizesc** [comment]
:+1: thanks guys

Looking forward to `1.0.0`
